File: week-02/day-2/src/ListIntroduction1.java
import java.util.ArrayList;

public class ListIntroduction1 {
    public static void main(String[] args) {
        // List introduction 1
        // We are going to play with lists.
        // Feel free to use the built-in methods where possible.
        //
        // Create an empty list which will contain names (strings)
        ArrayList<String> names = new ArrayList<>();
        System.out.println();

        System.out.println("Print out the number of elements in the list");
        System.out.println(names.size());
        System.out.println();

        // Add William to the list
        names.add("William");

        System.out.println("Print out whether the list is empty or not");
        System.out.println(names.isEmpty());
        System.out.println();

        // Add John to the list
        names.add("John");

        // Add Amanda to the list
        names.add("Amanda");

        System.out.println("Print out the number of elements in the list");
        System.out.println(names.size());
        System.out.println();

        System.out.println("Print out the 3rd element");
        System.out.println(names.get(2));
        System.out.println();

        System.out.println("Iterate through the list and print out each name");
        //      William
        //      John
        //      Amanda
        for (String name : names) {
            System.out.println(name);
        }
        System.out.println();

        System.out.println("Iterate through the list and print with numbers");
        //      1. William
        //      2. John
        //      3. Amanda
        for (int i = 0; i < names.size(); i++) {
            System.out.print(i + 1 + ". ");
            System.out.println(names.get(i));
        }
        System.out.println();

        // Remove the 2nd element
        names.remove(1);

        System.out.println("Iterate through the list in a reversed order and print out each name");
        //      Amanda
        //      William
        for (String name : names) {
            System.out.println(name);
        }
        System.out.println();

        // Remove all elements
        names.clear();
    }
}

File: python_solutions/week-02/day-2/data_structures/list_introduction_1.py
# # List introduction 1
#
# We are going to play with lists. Feel free to use the built-in methods where
# possible.
#
# - Create an empty list which will contain names (strings)
names = list()

# - Print out the number of elements in the list
print(len(names))

# - Add William to the list
names.append("William")

# - Print out whether the list is empty or not
if names:
    print("not empty")
else:
    print("empty")

# - Add John to the list
names.append("John")

# - Add Amanda to the list
names.append("Amanda")

# - Print out the number of elements in the list
print(len(names))

# - Print out the 3rd element
print(names[2])

# - Iterate through the list and print out each name
#   ```text
#   William
#   John
#   Amanda
#   ```
for name in names:
    print(name)

# - Iterate through the list and print
#   ```text
#   1. William
#   2. John
#   3. Amanda
#   ```
for i in range(len(names)):
    print("{}. {}".format(i + 1, names[i]))

# - Remove the 2nd element
names.pop(1)

# - Iterate through the list in a reversed order and print out each name
#   ```text
#   Amanda
#   William
#   ```
for name in names[::-1]:
    print(name)

# - Remove all elements
names.clear()
